### DISCUSSION TITLE
File Input

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#310] [Component suggestions]
<img width="480" alt="visuele weergave van de file input component" src="https://github.com/user-attachments/assets/3e8acbe9-9684-4cd7-9953-801057cfc117">

We willen graag meer leren over de File Input component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1726579540429539)

---

## Naam

File Input

## Doel

Knop om één of meerdere bestanden te kiezen en te uploaden.

## Ook bekend als

File Upload

## Gerelateerde componenten

[Form Field](https://github.com/orgs/nl-design-system/discussions/311), [File](https://github.com/orgs/nl-design-system/discussions/347)

## Onderzoek

[<input type="file"> `input` HTML element with type `file` - MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file)
Documentatie van Mozilla Developers Network over `input` HTML element with type `file`.

[File upload - GOV.UK](https://design-system.service.gov.uk/components/file-upload/)
File upload component bij GOV.UK waarbij onder andere documentatie is geschreven met betrekking tot WCAG 2.2. richtlijnen en foutmeldingen. Er is ook een [link naar de File upload GitHub discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/49) waar meer informatie, onderzoek en aanpassingen worden gedeeld.

In dit [gesloten issue bij GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/37#issuecomment-1576748073) wordt opgemerkt dat tijdens een extern accessibility audit bleek dat men een drag & drop indicatie mist. Ze lijken hier vooralsnog niks mee te doen.

[File upload - HRMC](https://design.tax.service.gov.uk/hmrc-design-patterns/file-upload/)
File upload component bij HRMC waarbij een alternatief patroon ten opzichte van GOV.UK wordt beschreven waarbij men meerdere bestanden kan uploaden.

[File input - USWDS](https://designsystem.digital.gov/components/file-input/#known-issues)
File input component bij U.S. Web Design System en de 'know issues' die zijn genoteerd.

## Notities 

- Het is bij de native File Input mogelijk om een bestand op de button te slepen.
- Het label en de tekst van de File Input verschilt per browser.

**Comment by ux-fran** [comment]
In project Overheidsbrede portalen hebben wij in het kader van het Wmebv voorbeeldformulier een issue aangemaakt waar File Input een grote rol speelt. [zie File input voor het Wmebv voorbeeldfoprmulier issue voor project Overheidsbrede portalen](https://github.com/nl-design-system/overheidsbrede-portalen-community/issues/152)

**Comment by jeffreylauwers** [comment]
## Upload ontwerp met Voorbeeld thema

We hebben een ontwerp gemaakt op basis van de [File uploader component](https://carbondesignsystem.com/components/file-uploader/usage/) van IBM’s Carbon design system. Deze staat in het [Figma Schetsbook van NL Design System](https://www.figma.com/design/fy08SZpZmqx6ljLwvA3Woe/NLDS---Schetsboek?node-id=1710-40863&t=cpysTfnzbRwOybbj-4).

Het betreft eerder een patroon dan een component, maar het geeft wellicht wel wat context voor de component.

Dit ontwerp is bedoelt voor als men meerdere bestanden tergelijk kan uploaden. Om die reden wordt het in het ontwerp ook elke keer gesproken over 'bestanden'.

Dit ontwerp heeft een rol gespeeld in het prototype rondom een klachtenformulier. Een paar punten die we sindsdien hebben geleerd:

**Men heeft niet altijd een beeld bij het aantal MB's en de verschillende bestandstypen zoals doc, docx, etc.**
Je zou de informatie over MB's en bestandstypen kunnen weglaten. Maar als deze restricties er zijn kunnen hier foutmeldingen uit voortkomen. Krijgt men dan niet het gevoel van "Had mij dat dan eerder verteld"?

**Men wil graag controleren wat zij hebben geupload.**
Daarom hebben we de bestandsnaam interactief gemaakt. Bij het gebruikersonderzoek rondom het klachtenformulier werd zelfs benoemd dat men graag een kleine thumbnail zou willen zien. Dit is nog niet verwerkt.

**Ook handig**
[Schrijfwijze grootheden en eenheden zoals MB en kB.](https://taaladvies.net/symbolen-voor-grootheden-en-eenheden-algemeen/)

<img width="1440" alt="Uploaden" src="https://github.com/user-attachments/assets/a8e32579-601d-4f62-8610-f6e8d15131dc">

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-file-input--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29186)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/file-input)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-23 om 15 58 25](https://github.com/user-attachments/assets/f161dbb4-9400-4720-909f-1df9426fde98)

Multiple Files
![Scherm­afbeelding 2024-09-23 om 16 01 42](https://github.com/user-attachments/assets/0dd31c38-5c68-4439-8cb6-7f0b175b67d1)

Disabled
![Scherm­afbeelding 2024-09-23 om 15 58 49](https://github.com/user-attachments/assets/fbeb1e35-2c5e-47b8-8edf-741535f3ac07)

---

### Notities

- Button en beschrijving staan binnen een container met rechte hoeken en een dotted border.
- Beschrijving naast de button wordt ingekort als hier in de breedte geen ruimte voor is.
- Mogelijkheid beschikbaar om te wisselen tussen acceptatie van enkele (Default) en meerdere ('Multiple') bestanden.
- Interessant stukje documentatie in Storybook.
> The filename label and button are displayed in the language of the browser and can vary between browsers and operating systems.

Dit resulteert in het volgende:

**Tekst aanpassingen per browser**

Chrome
![Scherm­afbeelding 2024-10-15 om 12 09 10](https://github.com/user-attachments/assets/c6df2e70-dc43-40fe-b8a3-69967eaf607d)

Safari
<img width="531" alt="Scherm­afbeelding 2024-10-15 om 12 08 48" src="https://github.com/user-attachments/assets/f334aef2-7fc8-4b20-98a2-fa56d003f1e2">

Firefox
![Scherm­afbeelding 2024-10-15 om 12 08 56](https://github.com/user-attachments/assets/70a86e48-873a-4607-945f-05e0461642a4)

Daarnaast verschilt ook de weergave van geüploade bestanden per browser. Dit geldt zowel bij het uploaden van één bestand als bij meerdere bestanden.

Chrome
<img width="680" height="118" alt="image" src="https://github.com/user-attachments/assets/8f26f503-626e-4255-acde-a436260fbc22" />

<img width="680" height="121" alt="image" src="https://github.com/user-attachments/assets/3e89ac78-a47a-4d63-a971-3a7a012ce0e0" />

Safari
<img width="680" height="123" alt="image" src="https://github.com/user-attachments/assets/6baf1b2e-dcea-4d2c-a6ff-304e429037c9" />

<img width="680" height="120" alt="image" src="https://github.com/user-attachments/assets/2b977adc-0b4f-4deb-8419-bd781d99490f" />

Firefox
<img width="680" height="119" alt="image" src="https://github.com/user-attachments/assets/7f6cb3e6-00de-44db-889c-589be16c21d2" />

<img width="680" height="119" alt="image" src="https://github.com/user-attachments/assets/35322c67-d8d2-4bc5-8179-66fdd8c10401" />

**Comment by ux-fran** [comment]
## Den Haag

- [Website Den Haag](https://formulier.denhaag.nl/Tripleforms/formulier/nl-NL/DefaultEnvironment/scvraagaandegemeente.aspx/fVraagAanDeGemeente)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=10817-61351&t=EIJasCsOjUWLHRVU-4)
---

### Screenshots / Varianten

**Website Den Haag**

Geen bestand gekozen
![afbeelding](https://github.com/user-attachments/assets/09f7ce91-158d-418c-a80e-02ba888f9405)

Wel een bestand gekozen
![afbeelding](https://github.com/user-attachments/assets/b6dc61f1-c954-48b0-88ba-94048447e413)

**Figma - Huidig**

Not inserted - inactive
![Scherm­afbeelding 2024-10-14 om 15 42 07](https://github.com/user-attachments/assets/6d4004e3-d6e0-44e5-a9d3-155c41ee5039)

Not inserted - active
![Scherm­afbeelding 2024-10-14 om 15 42 12](https://github.com/user-attachments/assets/84c0bf6f-2ccf-4e21-823a-d4e323f2301b)

Not inserted - hover
![Scherm­afbeelding 2024-10-14 om 15 42 16](https://github.com/user-attachments/assets/b84f5cfa-64b0-40b7-840c-b4dc9746f6b9)

Not inserted - focus
![Scherm­afbeelding 2024-10-14 om 15 42 22](https://github.com/user-attachments/assets/f20086da-5952-4112-9082-997baa9019e6)

Inserted - inactive
![Scherm­afbeelding 2024-10-14 om 15 42 31](https://github.com/user-attachments/assets/a9a9ffc8-1f02-42d5-be62-f828e0ca5483)

Inserted - error
![Scherm­afbeelding 2024-10-14 om 15 42 36](https://github.com/user-attachments/assets/70fca398-d09b-4df8-91e2-9fe6c7a81d34)

Met ingevulde teksten
![Scherm­afbeelding 2024-10-14 om 15 46 20](https://github.com/user-attachments/assets/65c465b9-7f95-4326-8d61-55e199916e71)

---
### Notities

**Website Den Haag**
- Tip bovenaan het blok, gemarkeerd met een *.
- Hulptekst met welke bestandstypen geüpload mogen worden.
- Button met label 'Bladeren'.
- Achter de button de tekst 'Geen bestanden geselecteerd' of niets als er wel een bestand geselecteerd is.
- Hulptekst onder de button "Er zijn nog geen bestanden ge-upload) of naam en grootte van het bestand .
- Toegevoegde bestanden worden onder de button geplaatst.

**Figma - Huidig**
- Onder de naam: File upload.
- Secondairy button met label 'Kies bestand'.
- Boven button staat de hulptekst 'Sleep uw bestanden hier of'.
- Onder button staat de criteria 'Max. bestandsgrootte 4MB'.
- Het geheel staat binnen een groot vlak met afgeronde hoeken en een solid border.
- Toegevoegde bestanden worden onder het vlak geplaatst doormiddel van [de File component](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=2334-3&t=EIJasCsOjUWLHRVU-4).

**Comment by ux-fran** [comment]
### DSO

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-files--files)

--- 

**Storybook**

Small viewport
![Scherm­afbeelding 2024-10-14 om 15 27 01](https://github.com/user-attachments/assets/22c632a7-413e-4e40-b50c-e95deb44fa5e)

Large viewport
![Scherm­afbeelding 2024-10-14 om 15 27 10](https://github.com/user-attachments/assets/58600759-61e2-4298-8ee2-b3b4bc58398b)


---
## Notities
- Primary button met plus icoon vooraf aan het label 'Document toevoegen'.
- Toegevoegde bestanden worden boven de button geplaatst.
- Per bestand:
  - Naam + extensie.
  - Een checkbox beschikbaar om vertrouwelijkheid aan te geven.
  - Een icoon met uitroepteken om aan te geven welke bestanden vertrouwelijk behandeld moeten worden.
  - Een prullenbak icoon om te kunnen verwijderen.
- Geel informatieblok met hulptekst + icoon over vertrouwelijkheid van bestanden.

**Reply by tfrijsewijk** [reply]
De DSO Toolkit maakt gebruik van `label + input[type="file"]` voor het interactie patroon:

![image](https://github.com/user-attachments/assets/f35388cc-32d0-4c26-81b3-d3899eaf3b0a)

Het label wordt gestijld als een button (en de focus styling wordt nagemaakt) en de click wordt doorgegeven. Hierdoor hoeven we geen scriptende `inputfileElement.click()` te doen als er op een echte native `<button>` wordt geklikt.

**Comment by ux-fran** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/file-upload)

---

### Screenshots / Varianten

**Documentatie website**

Basic
![Scherm­afbeelding 2024-10-14 om 15 35 50](https://github.com/user-attachments/assets/cb6412c1-a1ac-4cd9-ad71-2d23d013ce6f)

Met bestanden
![Scherm­afbeelding 2024-10-14 om 15 35 56](https://github.com/user-attachments/assets/9c4a07ec-d236-4339-81cd-5989c36778da)

Statussen (met bestanden)
![Scherm­afbeelding 2024-10-14 om 15 36 05](https://github.com/user-attachments/assets/44838b8e-dd11-4a03-8cce-692340996cd0)

Statussen (Te groot bestand)
![Scherm­afbeelding 2024-10-14 om 15 36 10](https://github.com/user-attachments/assets/7b8a2ede-c14c-4968-a4a2-cc3cc83c727c)

---

### Notities
- Onder de naam: File upload.
- Tekst 'Bestand uploaden' staat in een groot lichtblauw vlak met afgeronde hoeken en een dashed border.
- Boven het vlak staat het label 'Bestand uploaden'
- Onder het label (en boven het vlak) staan de criteria voor bestandsgrootte en bestandstype.
- Toegevoegde bestanden worden onder het vlak geplaatst doormiddel van [de File component](https://uno.dfront.rijkscloud.nl/#/components/file).
- Het lijkt erop dat als men 1 bestand toevoegd, de foutmelding boven het vlak wordt geplaatst.

**Comment by ux-fran** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-fileupload--happy-flow)

---
### Screenshots / Varianten

**Storybook**
FileUpload geen bestand gekozen
![afbeelding](https://github.com/user-attachments/assets/aa421014-c8b4-46f2-9c17-d40cd5591e8e)

FileUpload wel bestand gekozen
![afbeelding](https://github.com/user-attachments/assets/255f79c4-78f0-4700-8464-f7770c8c89ca)

Foutmelding - Alleen PDF
<img width="1128" alt="Scherm­afbeelding 2024-10-15 om 08 59 13" src="https://github.com/user-attachments/assets/128b21bd-27e9-4095-a574-2af4781a7875">

Foutmelding - Alleen PDF of PNG
<img width="1124" alt="Scherm­afbeelding 2024-10-15 om 09 02 09" src="https://github.com/user-attachments/assets/f47100d1-aaff-4b3b-a218-c6e5e5350a87">

Foutmelding - Virus
<img width="1133" alt="Scherm­afbeelding 2024-10-15 om 08 58 39" src="https://github.com/user-attachments/assets/c758b7e5-6548-4640-adc7-f0ebcd63df3c">


---

### Notities

- Onder de naam: File upload.
- Onder het label staat een tabel.
- Tabel met header met cellen voor icoon, File Name en Size kolommen.
- Indien er geen bestand is gekozen volgt er onder de tabel een vlak met rechte hoeken en dashed 3px border. Hierbinnen staat de tekst 'Drop a file, or browse to attach a file. maxFileSizeMessage'. Voorafgaand aan deze tekst staat een upload icoon (wolk met pijl omhoog).
- Indien er wel een bestand is gekozen wordt de tabel gevuld. Cel 1: prullenbak icoon, cel 2: bestandsnaam+ extensie, cel 3: bestandsgrootte.
- In de basis is er altijd een hulptekst 'Klik op een bestand om te selecteren en in te zenden' op een lichtblauw vlak. Maar deze kan aangevuld worden wanneer er een foutmelding is.

**Comment by ux-fran** [comment]
## OpenGemeenten

[Patternlab, iets naar beneden scrollen](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search/index.html)

---

### Screenshots / Varianten

**Patternlab**

Diverse varianten zonder gekozen bestand
![afbeelding](https://github.com/user-attachments/assets/125acd70-cf14-4ad4-995d-035b1f3d6da8)

Diverse varianten met gekozen bestand
![afbeelding](https://github.com/user-attachments/assets/e8dc3306-39c7-4efc-b1af-48f343b5a0f7)

---

### Notities

- Onder de naam: File upload.
- Label met indien gewenst '(Verplicht)'.
- Button om bestand te kiezen. Indien geen bestand geselecteerd de tekst: 'Geen bestand geselecteerd' en anders de bestandsnaam met extensie.
- Mogelijkheid voor een extra beschrijving onder de button.
- Foutmelding toont links een rode border voor de gehele component en tekst met rode letters onder de button.

**Comment by ux-fran** [comment]
## Rotterdam

[Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=3889-8381&t=XgMhMMUxWjzo7Y2R-4)

---
### Screenshots / Varianten

**Figma**

Single file upload
![Scherm­afbeelding 2024-10-15 om 10 37 59](https://github.com/user-attachments/assets/0c28ade7-fb53-47d2-8a6d-abc5e3437477)

File Dropper / multi file upload

Component
![Scherm­afbeelding 2024-10-15 om 10 37 08](https://github.com/user-attachments/assets/f5612f6f-98cb-45db-8125-ba3f6fb5d4cf)

Gedrag rondom 'File Dropper' bij 'Desktop & Tablet'

1. Button als trigger van Modal Dialog
![Scherm­afbeelding 2024-10-15 om 10 44 30](https://github.com/user-attachments/assets/a4e88b68-4598-410b-8eb5-87091667fa68)

2. Modal Dialog met File Dropper
![Scherm­afbeelding 2024-10-15 om 10 44 36](https://github.com/user-attachments/assets/73ac2d97-243d-4a46-8c82-8a69e31638de)

3. Modal Dialog met File Dropper, toegevoegde bestanden en button.
![Scherm­afbeelding 2024-10-15 om 10 44 43](https://github.com/user-attachments/assets/3f7b29d6-de01-4808-a324-0cadf879cbc1)

Foutmelding
![Scherm­afbeelding 2024-10-15 om 10 45 06](https://github.com/user-attachments/assets/128bb226-afe5-43a3-95b5-c79c0b65424b)

4. Toegevoegde bestanden worden getoond met optie om meer bestanden toe te voegen
![Scherm­afbeelding 2024-10-15 om 10 44 58](https://github.com/user-attachments/assets/1a844f3e-442d-4e73-8792-34b8409b3f2d)

Gedrag rondom 'File Dropper' bij 'Mobile'
![Scherm­afbeelding 2024-10-15 om 10 51 28](https://github.com/user-attachments/assets/2dcd8699-b228-4686-b19e-ff00798fe469)

---

### Notities

**Single file upload**
- Op basis van inputfield met placeholder tekst 'Kies een bestand'.
- Links staat een icoon van een document.
- Rechts staat een groene button met als label 'Kiezen'.

**Trigger File dropper**
- De File dropper component bevindt zich binnen een Modal Dialog. De Modal Dialog wordt geopend door de groene (secondary) button.
- De button toont een upload icoon en het label 'Bestanden toevoegen'.
- Onder de button staan de criteria 'Bestanden mogen pdf, Word, Excel of jpg (afbeelding) zijn en niet groter dan 20 MB per stuk.'

**Modal met File dropper**
- Modal heeft de titel de tekst 'Bestanden toevoegen'.
- Modal heeft als paragraaf de tekst 'Upload hier een pdf, jpg (afbeelding), Word- of Excel document.'

**File dropper**
- Tekst 'Klik om bestanden te kiezen of sleep meerdere bestanden hiernaartoe.' staat in een groot vlak met rechte hoeken en een dashed border.
- Van deze tekst is 'Klik om bestanden te kiezen' als link gemarkeerd.
- Onder deze tekst staat de criteria 'Maximaal 20 MB per bestand.
- Foutmelding wordt onder dit vlak getoond.
- Toegevoegde bestanden worden onder dit vlak getoond.

**Comment by ux-fran** [comment]
## RVO

**File input**
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-file-input--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=174-1813)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-fileinput)

**File input field**
- [Documentatie website - File input field](https://nl-design-system.github.io/rvo/docs/components/form-field-fileinput/)
- [Storybook - File input field](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-file-input-field--documentatie)

**Multi file upload**
[Figma - Multi file upload](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=508-25067)

**Patroon bestanden uploaden**
[Documentatie website - Patroon: Bestanden uploaden](https://nl-design-system.github.io/rvo/docs/patterns/gegevens-uitvragen/file-upload/)

---

### Screenshots / Varianten

**Documentatie website**
![Scherm­afbeelding 2024-10-15 om 11 26 39](https://github.com/user-attachments/assets/0b6778ef-a128-4321-b6e3-2abfd084d683)

**Storybook**

Default
![Scherm­afbeelding 2024-10-15 om 11 25 15](https://github.com/user-attachments/assets/a1e46885-c2c0-45b6-9f28-f43e16221ef8)

Focus
![Scherm­afbeelding 2024-10-15 om 11 25 27](https://github.com/user-attachments/assets/4993f9a0-1cd9-484e-b472-e827ca74834e)

Disabled
![Scherm­afbeelding 2024-10-15 om 11 25 20](https://github.com/user-attachments/assets/6f73d5cc-c59a-489a-963b-207d274bec5d)

Invalid
![Scherm­afbeelding 2024-10-15 om 11 25 33](https://github.com/user-attachments/assets/59bd6f96-9e1c-491e-88ba-9616de846565)

**Figma**
File input
![afbeelding](https://github.com/user-attachments/assets/0c35b550-112d-40bd-afca-8f2ea4a2aefa)

Multiple file upload
![Scherm­afbeelding 2024-10-15 om 11 49 16](https://github.com/user-attachments/assets/8ef350dc-95d6-4355-bc13-d4bc7ae87e30)

---

### Notities

**File input**
- De teksten 'Bestanden kiezen' en 'Geen bestand gekozen' staan binnen een vlak met rechte hoeken en een 2px solid border.
- Mogelijkheid beschikbaar om te wisselen tussen acceptatie van enkele (Default) en meerdere (Multiple) bestanden. Hierbij veranderd ook de tekst van Bestanden kiezen naar Bestand kiezen.
- In de documentatie staat het volgende:
> Toon altijd bestandsrestricties (formaat, extensies, maximale bestandsgroote etc.) onder de upload knop.

**Tekst aanpassingen per browser**

Chrome
![Scherm­afbeelding 2024-10-15 om 11 25 15](https://github.com/user-attachments/assets/0604323a-4314-4fc9-9f1f-fedff69d0ba9)

Safari
<img width="294" alt="Scherm­afbeelding 2024-10-15 om 12 05 12" src="https://github.com/user-attachments/assets/26439255-fd01-4159-8bd2-43e7f318c708">

Firefox
![Scherm­afbeelding 2024-10-15 om 12 05 29](https://github.com/user-attachments/assets/f7a4193c-cd24-46d6-acb9-055fdb1bb763)

**Comment by MariekeBrouwer** [comment]
![Mobile](https://github.com/user-attachments/assets/dfa34008-b6a2-471d-9f77-4bb34d409c74)
Groningen gebruikt documenten in setting van een formulier of in de Mijn Omgeving. Inwoner wordt dan gevraagd een document up te loaden wat nog ontbreekt (bijvoorbeeld een kopie van de bankpas of een verklaring van dokter bij aanvraag van (bijzondere) bijstand).

**Reply by UXTeamGroningen** [reply]
https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/mijn-zaken/groningen-mijn-omgeving/
Dit design was het resultaat van een onderzoek naar het Mijn Zaken onderdeel, waar het bestand uploaden een rol heeft.

Hieruit blijkt dat het document raster goed wordt begrepen en gebruikt. 
Bij de foutmelding van bijvoorbeeld een foutief bestand of te groot bestand wordt ook de melding begrepen en het prullenbak-icoon gebruikt om het document te delen.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[File Input op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=file+input)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[File Input op het Candidate bord](https://github.com/orgs/nl-design-system/projects/80/views/1?filterQuery=File+Input)
